### PR TITLE
Accessor

### PR DESCRIPTION
**Pull request by bdestombe**
- [x] Tests
- [x] Examples

- [x] Format

Leaving for a next PR:
- Clean up/remove old code
- Look for the TODO tags in the code
- Version bump
- Docs

### CODE REVIEW

**Comment by bdestombe** [comment]
@BSchilperoort This PR got a bit big and simultaneous you worked on many files in a different branch. What is a good moment to rebase? Which PR should be merged first?

**Comment by bdestombe** [comment]
Closes #200, #132, #127, #126

**Comment by BSchilperoort** [comment]
Nice work! :rocket:  I'll try to find some time to review it this week.

File: src/dtscalibration/dts_accessor.py [hunk 1]
```diff
@@ -1,161 +1,67 @@
...
+
+        # None if doesn't exist
+        self.st = xarray_obj.get("st")
+        self.ast = xarray_obj.get("ast")
+        self.rst = xarray_obj.get("rst")
+        self.rast = xarray_obj.get("rast")
+
+        self.acquisitiontime_fw = xarray_obj.get("userAcquisitionTimeFW")
+        self.acquisitiontime_bw = xarray_obj.get("userAcquisitionTimeBW")
         pass
```
Comment (BSchilperoort) [inline src/dtscalibration/dts_accessor.py:44]:
```suggestion
```

File: tests/test_examples.py [hunk 2]
```diff
@@ -1,72 +1,70 @@
...
-            out_path,
-            "--to",
-            "notebook",
-            "--execute",
-            "--ExecutePreprocessor.timeout=60",
-        ]
-        subprocess.check_call(args)
 
-        assert os.path.exists(nbpath), "nbconvert used different output filename"
+@pytest.mark.xfail
```
Comment (BSchilperoort) [inline tests/test_examples.py:24]: Nice to test this too! Should avoid any false negatives :)

**Review by BSchilperoort — APPROVED** [review]
Thanks for making all these changes! 🚀  It's good to finally have moved over to the accessor structure.

It was a lot of code to go through. Some parts I skimmed over more, but the tests seem to not have changed too much so I believe everything still works. The notebooks also still work so that's an extra test suite.

I just have some relatively minor comments, feel free to merge once you have resolved these.

File: docs/notebooks/01Load_xml_measurement_files.ipynb
[the remark this answers is not in the document]
Comment (bdestombe) [reply docs/notebooks/01Load_xml_measurement_files.ipynb:1]: Added to this PR

File: docs/notebooks/04Calculate_variance_Stokes.ipynb [hunk 3]
```diff
@@ -35,6 +35,8 @@
     "warnings.simplefilter(\"ignore\")  # Hide warnings to avoid clutter in the notebook\n",
     "\n",
     "from dtscalibration import read_silixa_files\n",
+    "from dtscalibration.dts_accessor import DtsAccessor  # noqa: E402\n",
```
[the remark this answers is not in the document]
Comment (bdestombe) [reply docs/notebooks/04Calculate_variance_Stokes.ipynb:38]: Let's solve this in a separate PR, See #205

File: src/dtscalibration/__init__.py
[the remark this answers is not in the document]
Comment (bdestombe) [reply src/dtscalibration/__init__.py:1]: See #205

File: src/dtscalibration/calibrate_utils.py [hunk 4]
```diff
@@ -485,7 +482,7 @@ def calibration_single_ended_solver(  # noqa: MC0001
     return (p_sol, p_var, p_cov) if calc_cov else (p_sol, p_var)
 
 
-def calibration_double_ended_helper(
+def calibrate_double_ended_helper(
     self,
```
[the remark this answers is not in the document]
Comment (bdestombe) [reply src/dtscalibration/calibrate_utils.py:486]: According to the example `self` is perfectly fine to use (https://docs.xarray.dev/en/stable/internals/extending-xarray.html) Could you elaborate a bit?

[the remark this answers is not in the document]
Comment (bdestombe) [reply src/dtscalibration/calibrate_utils.py:486]: And since all tests are working, could you open an issue?

**Pull request merged**